src/concurrency/data_race.rs

```diff
@@ -205,13 +205,36 @@ enum AtomicAccessType {
 /// are all treated as writes for the purpose
 /// of the data-race detector.
 #[derive(Copy, Clone, PartialEq, Eq, Debug)]
-enum NaWriteType {
+pub enum NaReadType {
+    /// Standard unsynchronized write.
+    Read,
+
+    Retag,
```

Comment:
The doc comment for `Read` seems to have a copy-paste mistake, and please also add one for `Retag`.